test/component_test.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef COMPONENT_TEST_HPP
+#define COMPONENT_TEST_HPP
+
+#include "mocks/spa_communicator.hpp"
+#include <component.hpp>
+#include <messages/spa_subscription_reply.hpp>
+#include <messages/spa_subscription_request.hpp>
+
+class ExampleComponent : public Component
+{
+public:
+  ExampleComponent(Component::Com com = nullptr, LogicalAddress address = LogicalAddress(0, 0)) : Component(com, address) {}
+
+  virtual void appInit()
+  {
+    return;
+  }
+};
+
+class ComponentTest : public ::testing::Test
+{
+public:
+  virtual void SetUp()
+  {
+  }
+
+  std::shared_ptr<MockSpaCommunicator> spaCom = std::make_shared<MockSpaCommunicator>();
```

Comment:
These should all be initialized in `SetUp`. This is because they will be reinitialized between every test, this is important in case a test changes the state of one of them it may skew the results of the tests.